### PR TITLE
Build and deploy to Netlify workflow

### PR DESCRIPTION
**Pull request by andynog**
closes: #19
 
Added the build and deploy to Netlify Github Actions workflow. 
  - Builds the Jekyll website (with versions)
  - Deploy to Netlify
  - Can be manually triggered, or run on schedule (daily), or when merges to main happen

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Build and Deploy to Netlify
+# This workflow runs periodically builds (daily) to deploy the
+# CometBFT documentation website to Netlify
+on:
+  pull_request:
```
Comment (thanethomson) [inline .github/workflows/build.yml:5]:
```suggestion
  push:
    branches:
      - main
```

Comment (thanethomson) [reply .github/workflows/build.yml:5]: If you run this on `pull_request`, then it's going to deploy on every PR, which isn't what we want.

**Review by thanethomson — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (andynog) [reply .github/workflows/build.yml:5]: lgtm, thanks.

**Review by andynog — COMMENTED** [review]
Looks good to me now.

**Review by thanethomson — APPROVED** [review]

**Pull request merged**